Tests/MongoKittenTests/CRUDTests.swift

```diff
@@ -1,55 +1,180 @@
-//import NIO
-//import MongoKitten
-//import XCTest
-//
-//let dbName = "test"
-//
-//class RemoteDatabaseCRUDTests : XCTestCase {
+import NIO
+import MongoKitten
+import XCTest
+
+let dbName = "KittenTest"
+
+class CrudTests : XCTestCase {
+  struct DummyAccount: Codable {
+    static let collectionName = "DummyAccounts"
+
+    let _id: ObjectId
+    var name: String
+    var password: String
+    var age: Int
+    
+    init (Name name: String, Password password: String, Age age: Int) {
+      self._id = ObjectId()
+      self.name = name
+      self.password = password
+      self.age = age
+    }
+  }
+  
+  let settings = ConnectionSettings(
+    authentication: .auto(username: "admin", password: "Autimatisering1"),
+    authenticationSource: "admin",
+    hosts: [
+        .init(hostname: "localhost", port: 27017)
+    ],
+    targetDatabase: dbName,
+    applicationName: "Test MK6"
+  )
+  var mongo : MongoDatabase!
+
+  override func setUpWithError() throws {
+    mongo = try MongoDatabase.synchronousConnect(settings: settings)
+  }
+
+  override func tearDownWithError() throws {
+    try mongo.drop().wait()
+  }
+
+  func testCreateDummyAccount () throws {
+    let schema = mongo[DummyAccount.collectionName]
+    let startingCount = try schema.count().wait()
+
+    let dummyAccount = DummyAccount(Name: "Dum", Password: "my", Age: 69)
+    let _ = try schema.insertEncoded(dummyAccount).wait()
```

Comment:
`_ =` is sufficient. Also, the reply of `insertEncoded` which is `InsertReply` can and should be used to verify that the command is successful (ok == 1) and that the insertedCount (otherwise referred to as N) matches the input.

Reply:
Noteworthy: The insertedCount can be 3 with an input of 5, and be `ok == 1`. This means that the 2 entities couldn't be inserted, probably due to unique constraints.